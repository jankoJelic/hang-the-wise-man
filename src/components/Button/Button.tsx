function Button({ title = "", onClick = () => {} }) {
  return (
    <button
      className="bg-mainAction text-white w-full h-10 rounded-xl hover:bg-mainLight mb-2"
      onClick={onClick}
    >
      {title}
    </button>
  );
}

export default Button;
