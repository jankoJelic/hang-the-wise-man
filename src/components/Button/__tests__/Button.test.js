import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Button from "../Button";

const mockOnClick = jest.fn();

const tree = renderer
  .create(<Button onClick={mockOnClick} title="Submit" />)
  .toJSON();

it("renders correctly", () => {
  expect(tree).toMatchSnapshot();
});

describe("onClick", () => {
  it("fires correctly", () => {
    render(<Button onClick={mockOnClick} title="Submit" />);
    const button = screen.getByRole("button");
    fireEvent.click(button);
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});