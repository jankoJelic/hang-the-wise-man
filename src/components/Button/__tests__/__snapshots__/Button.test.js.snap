// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="bg-mainAction text-white w-full max-w-md h-10 rounded-xl hover:bg-mainLight mb-2 font-semibold "
  disabled={false}
  onClick={[MockFunction]}
>
  Submit
</button>
`;
